### PR TITLE
fix(datastore): implement pydantic; improve code; improve imports; async vs threads benchmark; high level functions

### PR DESCRIPTION
**Pull request by luabida**

### CODE REVIEW

File: mosqlient/datastore/climate.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+from mosqlient import types
+from mosqlient.requests import get_all_sync
+from mosqlient._utils import parse_params
+
+
+class Climate(BaseModel):
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+
+    @classmethod
+    def get(cls, start: str | date, end: str | date, **kwargs):
```
Comment (luabida) [inline mosqlient/datastore/climate.py:14]: @fccoelho should start and end date obligatory params to retrieve Climate data?

File: mosqlient/datastore/infodengue.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+from mosqlient import types
+from mosqlient.requests import get_all_sync
+from mosqlient._utils import parse_params
+
+
+class Infodengue(BaseModel):
+    @classmethod
+    def get(
+        cls,
+        disease: Literal["dengue", "zika", "chikungunya"],
```
Comment (luabida) [inline mosqlient/datastore/infodengue.py:15]: Should `disease`, `start`, `end` parameters be obligatory to get Infodengue data?

**Comment by luabida** [comment]
Some benchmarking:

```py
import pyinstrument
import statistics

from mosqlient.datastore import Climate


def benchmark(bench):
    for i in range(1, 29):
        day, next_day = f"{i:02d}", f"{i+1:02d}"
        Climate.get(start=f"2022-01-{day}", end=f"2022-01-{next_day}",
                    geocode=3304557, env="dev", bench=bench)


def run_benchmark_async(iterations):
    profiler = pyinstrument.Profiler()
    runtimes = []

    for _ in range(iterations):
        profiler.start()
        benchmark(bench="a")
        profiler.stop()
        runtimes.append(profiler.last_session.duration)

    average_runtime = statistics.mean(runtimes)
    return average_runtime


def run_benchmark_threads(iterations):
    profiler = pyinstrument.Profiler()
    runtimes = []

    for _ in range(iterations):
        profiler.start()
        benchmark(bench="t")
        profiler.stop()
        runtimes.append(profiler.last_session.duration)

    average_runtime = statistics.mean(runtimes)
    return average_runtime



iteration_counts = [10, 100, 1000, 10000]

results_threads = {}
for count in iteration_counts:
    avg_runtime = run_benchmark_threads(count)
    results_threads[count] = avg_runtime
    print(
        f"Average runtime for {count} threads iterations: ",
        f"{avg_runtime:.6f} seconds"
    )

results_async = {}
for count in iteration_counts:
    avg_runtime = run_benchmark_async(count)
    results_async[count] = avg_runtime
    print(
        f"Average runtime for {count} async iterations: ",
        f"{avg_runtime:.6f} seconds"
    )
```

Average runtime for 10 threads iterations:  1.712777 seconds
Average runtime for 100 threads iterations:  16.481850 seconds
Average runtime for 1000 threads iterations:  162.960438 seconds
Average runtime for 10000 threads iterations:  1692.066486 seconds
Average runtime for 10 async iterations:  1.724071 seconds
Average runtime for 100 async iterations:  16.852182 seconds
Average runtime for 1000 async iterations:  192.103172 seconds
Average runtime for 10000 async iterations:  1725.047077 seconds

**Review by fccoelho — APPROVED** [review]
Looks good. Now we need an issu for documenting this simpler API

**Pull request merged**